doc/good-practices.mld

```diff
@@ -64,24 +64,61 @@ children. Some helpers for this are provided in {{!Ppxlib.Merlin_helpers}[Merlin
 
 {1:handling_errors Handling Errors}
 
-In order to give a nice user experience when reporting errors or failures in a
-PPX, it is necessary to include as much generated content as possible.
-Most IDE tools, such as Merlin, rely on the AST for their features, such as
-displaying type, jumping to definition, or showing the list of errors.
+In order to give a nice user experience when using a PPX, it is necessary that
+the resulting parsetree is as complete as possible.  Most IDE tools, such as
+Merlin, rely on the AST for their features, such as displaying type, jumping to
+definition, or showing the list of errors.
 
-{2 Embedding the Errors in the AST}
+In order to achieve this, errors that happen during rewriting should be handled
+in a way that do not prevent a meaningful AST to be passed to Merlin.
+
+There are mainly two ways to report errors when writing a PPX.
+
+- By embedding special extensions nodes, called "error nodes", inside the
+  generated code.
+- By raising a sepcific exception, letting the ppxlib driver handle the error.
+
+Let us emphasize that, while exceptions can be practical to quickly fail with an
+error, the embedding mechanism has many advantages. For instance, embedding
+allows to report multiple errors, and to output the part of the code that could
+be generated successfully.
+
+{2 Ppxlib and exceptions}
+
+In general, raising an exception in a registered transformation will make the
+ppxlib driver crash with an uncaught exception error. However, when spawned with
+the [-embed-errors] or [-as-ppx] flags (that's the case when Merlin calls the
+driver) the ppxlib driver still handles a specific kind of exception: Located
+exceptions. They have type {{!Ppxlib.Location.exception-Error}[Location.Error]}
+and contain enough information to display a located error message.
+
+During its {{!page-driver.driver_execution}execution}, the driver will run many
+different rewriter. In the case described above, it will catch any located
```

Comment:
```suggestion
During its {{!page-driver.driver_execution}execution}, the driver will run many
different rewriters. In the case described above, it will catch any located
```